libraries/Microsoft.Bot.Builder.Prompts/Choices/ChoiceFactory.cs

```diff
@@ -85,88 +91,106 @@ public static Activity Inline(List<Choice> choices, string text = null, string s
             };
 
             // Format list of choices
-            var connector = string.Empty;
-            var txt = text ?? string.Empty;
-            txt += " ";
-
-            for (var index = 0; index < choices.Count; index++)
+            var separator = string.Empty;
+            var sb = new StringBuilder();
+            if (text != null)
             {
-                var choice = choices[index];
-
-                var title = choice.Action != null && choice.Action.Title != null ? choice.Action.Title : choice.Value;
+                sb.Append(text);
+            }
+            sb.Append(" ");
 
-                txt += $"{connector}";
-                if (opt.IncludeNumbers.Value)
-                {
-                    txt += "(" + (index + 1).ToString() + ") ";
-                }
-                txt += $"{title}";
-                if (index == (choices.Count - 2))
-                {
-                    connector = (index == 0 ? opt.InlineOr : opt.InlineOrMore) ?? string.Empty;
-                }
-                else
+            if (choices != null)
+            {
+                int count = choices.Where(c => c != null).Count();
+                int index = 1;
+                foreach (var choice in choices.Where(c => c != null))
```

Comment:
I'm not sure how I feel about utilizing the `.Where(…)` to filter out potential null entries in the set. This feels to me like it should result in an exception and I would probably change the logic to something like:

```CSharp
if (choice == null)
{
    throw new ArgumentException("Set of choices contained a `null` instance which is not allowed.", nameof(choices));
}
```